airflow/jobs/scheduler_job.py

```diff
@@ -152,6 +152,43 @@ def __init__(
         self._zombie_threshold_secs = conf.getint("scheduler", "scheduler_zombie_task_threshold")
         self._standalone_dag_processor = conf.getboolean("scheduler", "standalone_dag_processor")
         self._dag_stale_not_seen_duration = conf.getint("scheduler", "dag_stale_not_seen_duration")
+
+        # Since the functionality for stalled_task_timeout, task_adoption_timeout, and worker_pods_pending_timeout
+        # are now handled by a single config (task_queued_timeout), we can't deprecate them as we normally would.
+        # So, we'll read each config and take the max value in order to ensure we're not undercutting a legitimate
+        # use of any of these configs.
+        stalled_task_timeout = conf.getfloat("celery", "stalled_task_timeout", fallback=0)
+        if stalled_task_timeout:
+            # TODO: Remove in Airflow 3.0
+            warnings.warn(
+                "The 'stalled_task_timeout' parameter is deprecated. "
+                "Please use 'scheduler.task_queued_timeout'.",
+                RemovedInAirflow3Warning,
+                stacklevel=2,
```

Comment:
Registering the deprecation at https://github.com/apache/airflow/blob/85427755f30d72e6c401ddc01b694b8a181c9a96/airflow/configuration.py#L175-L179 is enough. Airflow will automatically warn. No need to have another warning

Reply:
Based on [this comment](https://github.com/apache/airflow/pull/30375#discussion_r1157632498), I'm not sure it would make sense to deprecate in this manner in case one of these timeouts is intentionally set longer than whatever default is chosen for `task_queued_timeout`

Reply:
It might be worth iterating on this a bit, as ideally we'd get the old configs listed here so [this](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#base-log-folder-deprecated) still happens.

Maybe we get a "removed options" that doesn't do the "old first, then new" precedence when reading?